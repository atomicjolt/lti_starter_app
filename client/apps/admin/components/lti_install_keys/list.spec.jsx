import React from 'react';
import TestRenderer from 'react-test-renderer';
import List from './list';

describe('application instances list', () => {
  let result;
  let instance;

  const ltiInstallKeys = [
    {
      id: 4,
      application_id: 7,
      clientId: 'lti-key',
      iss: 'iss',
      jwksUrl: 'jwksUrl',
      tokenUrl: 'tokenUrl',
      oidcUrl: 'oidcUrl',
      created_at: 'created_at',
    },
  ];
  const application = {};
  const disableLtiInstallKey = () => {};
  const deleteLtiInstallKey = () => {};
  const saveLtiInstallKey = () => {};
  const currentSortColumn = '';
  const currentSortDirection = '';
  const setSort = () => {};

  beforeEach(() => {
    result = TestRenderer.create(<List
      ltiInstallKeys={ltiInstallKeys}
      application={application}
      disableLtiInstallKey={disableLtiInstallKey}
      deleteLtiInstallKey={deleteLtiInstallKey}
      saveLtiInstallKey={saveLtiInstallKey}
      currentSortColumn={currentSortColumn}
      currentSortDirection={currentSortDirection}
      setSort={setSort}
    />);
    instance = result.root;
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('renders table tags', () => {
    const thTags = instance.findAllByType('th');
    expect(thTags).toBeDefined();
    expect(thTags.length).toEqual(6);
  });
});
