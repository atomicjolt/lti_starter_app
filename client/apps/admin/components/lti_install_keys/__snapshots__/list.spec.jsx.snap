// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application instances list matches the snapshot 1`] = `
<table
  className="c-table c-table--instances"
>
  <thead>
    <tr>
      <th />
      <th
        className="sortable-header"
      >
        <div
          className="aj-flex"
        >
          <button
            className="aj-sort-btn"
            onClick={[Function]}
            type="button"
          >
            <span
              className="aj-flex"
            >
              ID
              <i
                className="material-icons aj-icon-btn-sort"
              >
                
              </i>
            </span>
          </button>
        </div>
      </th>
      <th
        className="sortable-header"
      >
        <div
          className="aj-flex"
        >
          <button
            className="aj-sort-btn"
            onClick={[Function]}
            type="button"
          >
            <span
              className="aj-flex"
            >
              ISS
              <i
                className="material-icons aj-icon-btn-sort"
              >
                
              </i>
            </span>
          </button>
        </div>
      </th>
      <th
        className="sortable-header"
      >
        <div
          className="aj-flex"
        >
          <button
            className="aj-sort-btn"
            onClick={[Function]}
            type="button"
          >
            <span
              className="aj-flex"
            >
              CLIENT ID
              <i
                className="material-icons aj-icon-btn-sort"
              >
                
              </i>
            </span>
          </button>
        </div>
      </th>
      <th
        className="sortable-header"
      >
        <div
          className="aj-flex"
        >
          <button
            className="aj-sort-btn"
            onClick={[Function]}
            type="button"
          >
            <span
              className="aj-flex"
            >
              CREATED
              <i
                className="material-icons aj-icon-btn-sort"
              >
                
              </i>
            </span>
          </button>
        </div>
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "grey",
              "cursor": "pointer",
              "fontSize": "1.5em",
            }
          }
          type="button"
        >
          <i
            className="material-icons"
          >
            settings
          </i>
        </button>
      </td>
      <td>
        4
      </td>
      <td>
        iss
      </td>
      <td>
        lti-key
      </td>
      <td>
        Invalid Date
         
        Invalid Date
      </td>
      <td>
        <button
          className="c-delete"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "grey",
              "cursor": "pointer",
              "fontSize": "1.5em",
            }
          }
          type="button"
        >
          <i
            className="i-delete"
          />
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
